File: app/src/main/java/com/reteno/sample/CustomPushReceiver.java
package com.reteno.sample;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.widget.Toast;

import com.reteno.core.util.Logger;

public class CustomPushReceiver extends BroadcastReceiver {

    public static final String TAG = CustomPushReceiver.class.getSimpleName();

    @Override
    public void onReceive(Context context, Intent intent) {
        Bundle extras = intent.getExtras();
        Toast.makeText(context, "Received custom push. Bundle.size = " + extras.size(), Toast.LENGTH_SHORT).show();
        /*@formatter:off*/ Logger.i(TAG, "onReceive(): ", "context = [" , context , "], intent = [", intent, "]");
        /*@formatter:on*/
    }

}

File: app/src/main/java/com/reteno/sample/CustomPushReceiver.kt
package com.reteno.sample

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.widget.Toast
import com.reteno.core.util.Logger.i

class CustomPushReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val extras = intent.extras
        Toast.makeText(
            context,
            "Received custom push. Bundle.size = " + extras!!.size(),
            Toast.LENGTH_SHORT
        ).show()
        /*@formatter:off*/i(TAG, "onReceive(): ", "context = [", context, "], intent = [", intent, "]")
            /*@formatter:on*/
    }

    companion object {
        private val TAG = CustomPushReceiver::class.java.getSimpleName()
    }
}
